packages/next/src/client/components/react-dev-overlay/_experimental/app/react-dev-overlay.tsx

```diff
@@ -62,12 +62,14 @@ export default function ReactDevOverlay({
   )
 
   return (
-    <DevOverlayErrorBoundary
-      devOverlay={devOverlay}
-      globalError={globalError}
-      onError={setIsErrorOverlayOpen}
-    >
-      {children}
-    </DevOverlayErrorBoundary>
+    <>
+      <DevOverlayErrorBoundary
+        globalError={globalError}
+        onError={setIsErrorOverlayOpen}
+      >
+        {children}
+      </DevOverlayErrorBoundary>
+      <Suspense>{devOverlay}</Suspense>
```

Comment:
The added Suspense boundary is a safety measure. Whatever is in `devOverlay` should really just throw during SSR to skip SSRing.